### DISCUSSION TITLE
Performance question in Win2D on WinUI 3 (win32)

### DISCUSSION THREAD

**Discussion by dimohy** [microsoft/WindowsAppSDK#1212] [Q&A]
I want to use Win2D in WinUI 3 (win32).

The 'Gallery of Win2D examples' from the Microsoft Store shows very beautiful behavior.

So, using the source code of 'Win2D Example Gallery', I understand Win2D and reconfigure it as a WinUI 3 (win32) project.

However, the performance of 'Invalidate()' in 'CanvasControl' is significantly different from Win2D in UWP.

I want the performance of Win2D on UWP. I'm not at the level to look into the source code of WinUI or Win2D, so I desperately need your answer.

![image](https://user-images.githubusercontent.com/58620778/128848120-d84f4918-d90b-4610-90c7-1f2792ce12ce.png)

**Comment by daamenik** [comment]
Hi dimohy, which examples are you experiencing the poor performance on?

**Reply by dimohy** [reply]
CanvasAnimatedControl is not yet supported by WinUI 3, so it happens in the example where CanvasControl is used. From what I have confirmed, it seems that it is because of the delay between drawing again after executing the canvas.Invalidate() method rather than drawing performance. What I suspect is that the fps seems to be locked.

Please refer to the following link.

https://github.com/microsoft/Win2D/issues/842
